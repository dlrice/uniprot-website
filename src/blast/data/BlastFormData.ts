export type BlastFormValue = {
  fieldName: string;
  selected?: string | undefined;
  selectedLabel?: string | undefined;
  type: BlastFieldTypes;
  values?: { label?: string; value?: string | number }[];
};

export enum BlastFields {
  program = 'Program',
  stype = 'Sequence type',
  email = 'Email',
  sequence = 'Sequence',
  taxon = 'Taxon',
  targetDb = 'Target database',
  threshold = 'E-Threshold',
  matrix = 'Matrix',
  filter = 'Filter',
  gapped = 'Gapped',
  hits = 'Hits',
}

export enum BlastFieldTypes {
  textarea,
  select,
  autocomplete,
}

export type BlastFormValues = { [x in BlastFields]: BlastFormValue };

export default {
  [BlastFields.program]: {
    fieldName: 'program',
    selected: 'blastp',
  },
  [BlastFields.stype]: {
    fieldName: 'stype',
    selected: 'protein',
  },
  [BlastFields.email]: {
    fieldName: 'email',
    selected: 'uuw_dev@uniprot.org',
  },
  [BlastFields.sequence]: {
    fieldName: 'sequence',
    type: BlastFieldTypes.textarea,
  },
  [BlastFields.targetDb]: {
    fieldName: 'database',
    type: BlastFieldTypes.select,
    selected: 'uniprotkb_refprotswissprot',
    values: [
      {
        value: 'uniprotkb_refprotswissprot',
        label: 'UniProtKb',
      },
    ],
  },
  [BlastFields.taxon]: {
    fieldName: 'taxids',
    type: BlastFieldTypes.autocomplete,
  },
  [BlastFields.threshold]: {
    fieldName: 'threshold',
    type: BlastFieldTypes.select,
    selected: '10',
    values: [
      { value: '0.0001' },
      { value: '0.001' },
      { value: '0.01' },
      { value: '0.1' },
      { value: '1' },
      { value: '10' },
      { value: '100' },
      { value: '1000' },
    ],
  },
  [BlastFields.matrix]: {
    fieldName: 'matrix',
    type: BlastFieldTypes.select,
    selected: null,
    values: [
      {
        value: '',
        label: 'Auto',
      },
      { value: '45' },
      { value: '62' },
      { value: '80' },
      { value: '70' },
      { value: '30' },
    ],
  },
  [BlastFields.filter]: {
    fieldName: 'filter',
    type: BlastFieldTypes.select,
    selected: 'F',
    values: [
      { value: 'F', label: 'None' },
      { valueNone: 'T', label: 'Filter low complexity regions' },
      { value: 'mask', label: 'Mask lookup table only' },
    ],
  },
  [BlastFields.gapped]: {
    fieldName: 'gapped',
    type: BlastFieldTypes.select,
    selected: 'true',
    values: [
      { value: 'true', label: 'yes' },
      { value: 'false', label: 'no' },
    ],
  },
  [BlastFields.hits]: {
    fieldName: 'hits',
    type: BlastFieldTypes.select,
    selected: '250',
    values: [
      { value: '50' },
      { value: '100' },
      { value: '250' },
      { value: '500' },
      { value: '750' },
      { value: '1000' },
    ],
  },
};
