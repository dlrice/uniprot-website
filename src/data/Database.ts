import { Database, DatabaseCategory } from '../model/types/databaseTypes';
import EntrySection from '../model/types/EntrySection';

export const databaseCategoryToString = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

const databaseCategoryToDatabases = new Map<DatabaseCategory, Database[]>();
databaseCategoryToDatabases.set(DatabaseCategory.SEQUENCE, [
  // Not found: EMBL/GenBank/DDBJ
  Database.EMBL,
  Database.CCDS,
  Database.PIR,
  Database.RefSeq,
  Database.UniGene,
]);
databaseCategoryToDatabases.set(DatabaseCategory.STRUCTURE, [
  Database.PDB,
  Database.SMR,
  Database.DisProt,
  Database.ProteinModelPortal,
]);
databaseCategoryToDatabases.set(DatabaseCategory.INTERACTION, [
  Database.BioGrid,
  Database.DIP,
  Database.IntAct,
  Database.MINT,
  Database.STRING,
]);
databaseCategoryToDatabases.set(DatabaseCategory.CHEMISTRY, [
  Database.BindingDB,
  Database.ChEMBL,
  Database.DrugBank,
  Database.GuidetoPHARMACOLOGY,
]);
databaseCategoryToDatabases.set(DatabaseCategory.FAMILY, [
  // Not found: Database.PptaseDB,
  Database.Allergome,
  Database.CAZy,
  Database.MEROPS,
  Database.MoonProt,
  Database.mycoCLAP,
  Database.PeroxiBase,
  Database.REBASE,
  Database.TCDB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PTM, [
  // Not found: DEPOD, PhosphoSite, PhosSite
  Database.DEPOD,
  Database.PhosphoSitePlus,
  Database.UniCarbKB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.POLYMORPHISM, [
  Database.dbSNP,
  Database.DMDM,
]);
databaseCategoryToDatabases.set(DatabaseCategory.GEL, [
  Database.SWISS_2DPAGE,
  Database.COMPLUYEAST_2DPAGE,
  Database.DOSAC_COBS_2DPAGE,
  Database.OGP,
  Database.REPRODUCTION_2DPAGE,
  Database.UCD_2DPAGE,
  Database.World_2DPAGE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTEOMIC, [
  Database.MaxQB,
  Database.PaxDb,
  Database.PeptideAtlas,
  Database.PRIDE,
  Database.ProMEX,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTOCOL, [Database.DNASU]);
databaseCategoryToDatabases.set(DatabaseCategory.GENOME, [
  Database.Ensembl,
  Database.EnsemblBacteria,
  Database.EnsemblFungi,
  Database.EnsemblMetazoa,
  Database.EnsemblPlants,
  Database.EnsemblProtists,
  Database.GeneCards,
  Database.GeneID,
  Database.KEGG,
  Database.PATRIC,
  Database.UCSC,
  Database.VectorBase,
]);
databaseCategoryToDatabases.set(DatabaseCategory.ORGANISM, [
  Database.ArachnoServer,
  Database.CGD,
  Database.ConoServer,
  Database.CTD,
  Database.dictyBase,
  Database.EchoBASE,
  Database.EcoGene,
  Database.euHCVdb,
  Database.EuPathDB,
  Database.FlyBase,
  Database.GeneReviews,
  Database.Gramene,
  Database.H_InvDB,
  Database.HGNC,
  Database.HPA,
  Database.LegioList,
  Database.Leproma,
  Database.MaizeGDB,
  Database.MIM,
  Database.MGI,
  Database.neXtProt,
  Database.Orphanet,
  Database.PharmGKB,
  Database.PomBase,
  Database.PseudoCAP,
  Database.RGD,
  Database.SGD,
  Database.TAIR,
  Database.TubercuList,
  Database.WormBase,
  Database.Xenbase,
  Database.ZFIN,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PHYLOGENOMIC, [
  Database.eggNOG,
  Database.GeneTree,
  Database.HOGENOM,
  Database.HOVERGEN,
  Database.InParanoid,
  Database.KO,
  Database.OMA,
  Database.OrthoDB,
  Database.PhylomeDB,
  Database.TreeFam,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PATHWAY, [
  Database.BioCyc,
  Database.BRENDA,
  Database.Reactome,
  Database.SABIO_RK,
  Database.SignaLink,
  Database.UniPathway,
]);
databaseCategoryToDatabases.set(DatabaseCategory.EXPRESSION, [
  Database.Bgee,
  Database.ExpressionAtlas,
]);
databaseCategoryToDatabases.set(DatabaseCategory.DOMAIN, [
  Database.HAMAP,
  Database.InterPro,
  Database.Gene3D,
  Database.PANTHER,
  Database.Pfam,
  Database.PIRSF,
  Database.PRINTS,
  Database.ProDom,
  Database.SMART,
  Database.SUPFAM,
  Database.TIGRFAMs,
  Database.PROSITE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.OTHER, [
  // Not found: NextBio
  Database.ChiTaRS,
  Database.EvolutionaryTrace,
  Database.GenomeRNAi,
  Database.GeneWiki,
  Database.PMAP_CutDB,
  Database.PRO,
]);

const entrySectionToDatabaseCategories = new Map<
  EntrySection,
  DatabaseCategory[]
>();
entrySectionToDatabaseCategories.set(EntrySection.Expression, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.EXPRESSION,
]);
entrySectionToDatabaseCategories.set(EntrySection.FamilyAndDomains, [
  DatabaseCategory.FAMILY,
  DatabaseCategory.PHYLOGENOMIC,
  DatabaseCategory.DOMAIN,
]);
entrySectionToDatabaseCategories.set(EntrySection.Function, [
  DatabaseCategory.PATHWAY,
  DatabaseCategory.FAMILY,
]);
entrySectionToDatabaseCategories.set(EntrySection.Interaction, [
  DatabaseCategory.INTERACTION,
  DatabaseCategory.CHEMISTRY,
]);
entrySectionToDatabaseCategories.set(EntrySection.NamesAndTaxonomy, [
  DatabaseCategory.ORGANISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.PathologyOrBioTech, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.OTHER,
  DatabaseCategory.FAMILY,
  DatabaseCategory.CHEMISTRY,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.ProteinProcessing, [
  DatabaseCategory.PROTEOMIC,
  DatabaseCategory.GEL,
  DatabaseCategory.PTM,
  DatabaseCategory.OTHER,
]);
entrySectionToDatabaseCategories.set(EntrySection.Sequence, [
  DatabaseCategory.SEQUENCE,
  DatabaseCategory.GENOME,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.Structure, [
  DatabaseCategory.STRUCTURE,
  DatabaseCategory.PROTOCOL,
  DatabaseCategory.OTHER,
]);

// Combine multiple maps into single map for quick read access.
export const entrySectionDatabaseToDatabaseCategory = new Map<
  string,
  DatabaseCategory
>();
for (const [
  entrySection,
  databaseCategories,
] of entrySectionToDatabaseCategories) {
  if (databaseCategories) {
    for (const databaseCategory of databaseCategories) {
      const databases = databaseCategoryToDatabases.get(databaseCategory);
      if (databases) {
        for (const database of databases) {
          entrySectionDatabaseToDatabaseCategory.set(
            `${entrySection}_${database}`,
            databaseCategory
          );
        }
      }
    }
  }
}

export default entrySectionDatabaseToDatabaseCategory;
