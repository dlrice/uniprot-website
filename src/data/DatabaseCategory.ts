enum databaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

export default databaseCategory;
