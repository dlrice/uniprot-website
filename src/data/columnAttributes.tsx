const small = 200; //10rem;
const medium = 400; //15rem;
const large = 600; //20rem;

export type columnAttributes = {
  [columnName: string]: {
    width: number;
    label: string;
  };
};

const columnAttributes: columnAttributes = {
  accession: {
    width: small,
    label: 'Entry',
  },
  id: {
    width: small,
    label: 'Entry Name',
  },
  protein_name: {
    width: large,
    label: 'Protein names',
  },
  gene_names: {
    width: medium,
    label: 'Gene Names',
  },
  organism: {
    width: large,
    label: 'Organism',
  },
};

export default columnAttributes;
