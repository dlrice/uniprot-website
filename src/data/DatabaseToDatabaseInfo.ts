import Database from './Database';

interface AttributesItem {
  name: string;
  xmlTag: string;
  uriLink?: string;
}

interface DatabaseInfo {
  name: string;
  displayName?: string;
  category: string;
  uriLink: string;
  attributes?: AttributesItem[];
}

interface DatabaseToDatabaseInfo {
  [database: string]: DatabaseInfo;
  [Database.EMBL]: DatabaseInfo;
  [Database.CCDS]: DatabaseInfo;
  [Database.PIR]: DatabaseInfo;
  [Database.RefSeq]: DatabaseInfo;
  [Database.UniGene]: DatabaseInfo;
  [Database.PDB]: DatabaseInfo;
  [Database.PDBsum]: DatabaseInfo;
  [Database.DisProt]: DatabaseInfo;
  [Database.ProteinModelPortal]: DatabaseInfo;
  [Database.SMR]: DatabaseInfo;
  [Database.BioGrid]: DatabaseInfo;
  [Database.ComplexPortal]: DatabaseInfo;
  [Database.CORUM]: DatabaseInfo;
  [Database.DIP]: DatabaseInfo;
  [Database.ELM]: DatabaseInfo;
  [Database.IntAct]: DatabaseInfo;
  [Database.MINT]: DatabaseInfo;
  [Database.STRING]: DatabaseInfo;
  [Database.BindingDB]: DatabaseInfo;
  [Database.ChEMBL]: DatabaseInfo;
  [Database.DrugBank]: DatabaseInfo;
  [Database.GuidetoPHARMACOLOGY]: DatabaseInfo;
  [Database.SwissLipids]: DatabaseInfo;
  [Database.Allergome]: DatabaseInfo;
  [Database.CAZy]: DatabaseInfo;
  [Database.ESTHER]: DatabaseInfo;
  [Database.IMGT_GENE_DB]: DatabaseInfo;
  [Database.MEROPS]: DatabaseInfo;
  [Database.MoonDB]: DatabaseInfo;
  [Database.MoonProt]: DatabaseInfo;
  [Database.mycoCLAP]: DatabaseInfo;
  [Database.PeroxiBase]: DatabaseInfo;
  [Database.REBASE]: DatabaseInfo;
  [Database.TCDB]: DatabaseInfo;
  [Database.UniLectin]: DatabaseInfo;
  [Database.CarbonylDB]: DatabaseInfo;
  [Database.DEPOD]: DatabaseInfo;
  [Database.GlyConnect]: DatabaseInfo;
  [Database.iPTMnet]: DatabaseInfo;
  [Database.PhosphoSitePlus]: DatabaseInfo;
  [Database.SwissPalm]: DatabaseInfo;
  [Database.UniCarbKB]: DatabaseInfo;
  [Database.BioMuta]: DatabaseInfo;
  [Database.DMDM]: DatabaseInfo;
  [Database.dbSNP]: DatabaseInfo;
  [Database.COMPLUYEAST_2DPAGE]: DatabaseInfo;
  [Database.DOSAC_COBS_2DPAGE]: DatabaseInfo;
  [Database.OGP]: DatabaseInfo;
  [Database.REPRODUCTION_2DPAGE]: DatabaseInfo;
  [Database.SWISS_2DPAGE]: DatabaseInfo;
  [Database.UCD_2DPAGE]: DatabaseInfo;
  [Database.World_2DPAGE]: DatabaseInfo;
  [Database.EPD]: DatabaseInfo;
  [Database.MaxQB]: DatabaseInfo;
  [Database.PaxDb]: DatabaseInfo;
  [Database.PeptideAtlas]: DatabaseInfo;
  [Database.PRIDE]: DatabaseInfo;
  [Database.ProMEX]: DatabaseInfo;
  [Database.ProteomicsDB]: DatabaseInfo;
  [Database.TopDownProteomics]: DatabaseInfo;
  [Database.jPOST]: DatabaseInfo;
  [Database.DNASU]: DatabaseInfo;
  [Database.Ensembl]: DatabaseInfo;
  [Database.EnsemblBacteria]: DatabaseInfo;
  [Database.EnsemblFungi]: DatabaseInfo;
  [Database.EnsemblMetazoa]: DatabaseInfo;
  [Database.EnsemblPlants]: DatabaseInfo;
  [Database.EnsemblProtists]: DatabaseInfo;
  [Database.GeneDB]: DatabaseInfo;
  [Database.GeneID]: DatabaseInfo;
  [Database.Gramene]: DatabaseInfo;
  [Database.KEGG]: DatabaseInfo;
  [Database.PATRIC]: DatabaseInfo;
  [Database.UCSC]: DatabaseInfo;
  [Database.VectorBase]: DatabaseInfo;
  [Database.WBParaSite]: DatabaseInfo;
  [Database.ArachnoServer]: DatabaseInfo;
  [Database.Araport]: DatabaseInfo;
  [Database.CGD]: DatabaseInfo;
  [Database.ConoServer]: DatabaseInfo;
  [Database.CTD]: DatabaseInfo;
  [Database.dictyBase]: DatabaseInfo;
  [Database.DisGeNET]: DatabaseInfo;
  [Database.EchoBASE]: DatabaseInfo;
  [Database.EcoGene]: DatabaseInfo;
  [Database.euHCVdb]: DatabaseInfo;
  [Database.EuPathDB]: DatabaseInfo;
  [Database.FlyBase]: DatabaseInfo;
  [Database.GeneCards]: DatabaseInfo;
  [Database.GeneReviews]: DatabaseInfo;
  [Database.H_InvDB]: DatabaseInfo;
  [Database.HGNC]: DatabaseInfo;
  [Database.HPA]: DatabaseInfo;
  [Database.LegioList]: DatabaseInfo;
  [Database.Leproma]: DatabaseInfo;
  [Database.MaizeGDB]: DatabaseInfo;
  [Database.MalaCards]: DatabaseInfo;
  [Database.MGI]: DatabaseInfo;
  [Database.MIM]: DatabaseInfo;
  [Database.neXtProt]: DatabaseInfo;
  [Database.OpenTargets]: DatabaseInfo;
  [Database.Orphanet]: DatabaseInfo;
  [Database.PharmGKB]: DatabaseInfo;
  [Database.PomBase]: DatabaseInfo;
  [Database.PseudoCAP]: DatabaseInfo;
  [Database.RGD]: DatabaseInfo;
  [Database.SGD]: DatabaseInfo;
  [Database.TAIR]: DatabaseInfo;
  [Database.TubercuList]: DatabaseInfo;
  [Database.VGNC]: DatabaseInfo;
  [Database.WormBase]: DatabaseInfo;
  [Database.Xenbase]: DatabaseInfo;
  [Database.ZFIN]: DatabaseInfo;
  [Database.eggNOG]: DatabaseInfo;
  [Database.GeneTree]: DatabaseInfo;
  [Database.HOGENOM]: DatabaseInfo;
  [Database.HOVERGEN]: DatabaseInfo;
  [Database.InParanoid]: DatabaseInfo;
  [Database.KO]: DatabaseInfo;
  [Database.OMA]: DatabaseInfo;
  [Database.OrthoDB]: DatabaseInfo;
  [Database.PhylomeDB]: DatabaseInfo;
  [Database.TreeFam]: DatabaseInfo;
  [Database.BioCyc]: DatabaseInfo;
  [Database.BRENDA]: DatabaseInfo;
  [Database.Reactome]: DatabaseInfo;
  [Database.SABIO_RK]: DatabaseInfo;
  [Database.SignaLink]: DatabaseInfo;
  [Database.SIGNOR]: DatabaseInfo;
  [Database.UniPathway]: DatabaseInfo;
  [Database.ChiTaRS]: DatabaseInfo;
  [Database.EvolutionaryTrace]: DatabaseInfo;
  [Database.GeneWiki]: DatabaseInfo;
  [Database.GenomeRNAi]: DatabaseInfo;
  [Database.PMAP_CutDB]: DatabaseInfo;
  [Database.PRO]: DatabaseInfo;
  [Database.Bgee]: DatabaseInfo;
  [Database.CleanEx]: DatabaseInfo;
  [Database.CollecTF]: DatabaseInfo;
  [Database.ExpressionAtlas]: DatabaseInfo;
  [Database.Genevisible]: DatabaseInfo;
  [Database.CDD]: DatabaseInfo;
  [Database.Gene3D]: DatabaseInfo;
  [Database.HAMAP]: DatabaseInfo;
  [Database.InterPro]: DatabaseInfo;
  [Database.PANTHER]: DatabaseInfo;
  [Database.Pfam]: DatabaseInfo;
  [Database.PIRSF]: DatabaseInfo;
  [Database.PRINTS]: DatabaseInfo;
  [Database.ProDom]: DatabaseInfo;
  [Database.SFLD]: DatabaseInfo;
  [Database.SMART]: DatabaseInfo;
  [Database.SUPFAM]: DatabaseInfo;
  [Database.TIGRFAMs]: DatabaseInfo;
  [Database.PROSITE]: DatabaseInfo;
  [Database.GO]: DatabaseInfo;
  [Database.Proteomes]: DatabaseInfo;
  [Database.GlycoProtDB]: DatabaseInfo;
  [Database.ProtClustDB]: DatabaseInfo;
  [Database.PathwayCommons]: DatabaseInfo;
}

export default DatabaseToDatabaseInfo;
