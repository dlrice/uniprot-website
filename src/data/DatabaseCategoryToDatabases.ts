import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string[]>([
  [
    DatabaseCategory.SEQUENCE,
    ['EMBL/GenBank/DDBJ', 'CCDS', 'PIR', 'RefSeq', 'UniGene'],
  ],
  [DatabaseCategory.STRUCTURE, ['PDB', 'SMR', 'DisProt', 'ProteinModelPortal']],
  [
    DatabaseCategory.INTERACTION,
    ['BioGrid', 'DIP', 'IntAct', 'MINT', 'STRING'],
  ],
  [
    DatabaseCategory.CHEMISTRY,
    ['BindingDB', 'ChEMBL', 'DrugBank', 'GuidetoPHARMACOLOGY'],
  ],
  [
    DatabaseCategory.FAMILY,
    [
      'Allergome',
      'CAZy',
      'MEROPS',
      'MoonProt',
      'mycoCLAP',
      'PeroxiBase',
      'PptaseDB',
      'REBASE',
      'TCDB',
    ],
  ],
  [DatabaseCategory.PTM, ['DEPOD', 'PhosphoSite', 'PhosSite', 'UniCarbKB']],
  [DatabaseCategory.POLYMORPHISM, ['dbSNP', 'DMDM']],
  [
    DatabaseCategory.GEL,
    [
      'SWISS-2DPAGE',
      'COMPLUYEAST-2DPAGE',
      'DOSAC-COBS-2DPAGE',
      'OGP',
      'REPRODUCTION-2DPAGE',
      'UCD-2DPAGE',
      'World-2DPAGE',
    ],
  ],
  [
    DatabaseCategory.PROTEOMIC,
    ['MaxQB', 'PaxDb', 'PeptideAtlas', 'PRIDE', 'ProMEX'],
  ],
  [DatabaseCategory.PROTOCOL, ['DNASU']],
  [
    DatabaseCategory.GENOME,
    [
      'Ensembl and Ensembl Genomes',
      'Ensembl',
      'EnsemblBacteria',
      'EnsemblFungi',
      'EnsemblMetazoa',
      'EnsemblPlants',
      'EnsemblProtists',
      'GeneCards',
      'GeneID',
      'KEGG',
      'PATRIC',
      'UCSC',
      'VectorBase',
    ],
  ],
  [
    DatabaseCategory.ORGANISM,
    [
      'ArachnoServer',
      'CGD',
      'ConoServer',
      'CTD',
      'dictyBase',
      'EchoBASE',
      'EcoGene',
      'euHCVdb',
      'EuPathDB',
      'FlyBase',
      'GeneReviews',
      'Gramene',
      'H-InvDB',
      'HGNC',
      'HPA',
      'LegioList',
      'Leproma',
      'MaizeGDB',
      'MIM',
      'MGI',
      'neXtProt',
      'Orphanet',
      'PharmGKB',
      'PomBase',
      'PseudoCAP',
      'RGD',
      'SGD',
      'TAIR',
      'TubercuList',
      'WormBase',
      'Xenbase',
      'ZFIN',
    ],
  ],
  [
    DatabaseCategory.PHYLOGENOMIC,
    [
      'eggNOG',
      'GeneTree',
      'HOGENOM',
      'HOVERGEN',
      'InParanoid',
      'KO',
      'OMA',
      'OrthoDB',
      'PhylomeDB',
      'TreeFam',
    ],
  ],
  [
    DatabaseCategory.PATHWAY,
    ['BioCyc', 'BRENDA', 'Reactome', 'SABIO-RK', 'SignaLink', 'UniPathway'],
  ],
  [DatabaseCategory.EXPRESSION, ['Bgee', 'ExpressionAtlas']],
  [
    DatabaseCategory.DOMAIN,
    [
      'HAMAP',
      'InterPro',
      'Gene3D',
      'PANTHER',
      'Pfam',
      'PIRSF',
      'PRINTS',
      'ProDom',
      'SMART',
      'SUPFAM',
      'TIGRFAMs',
      'PROSITE',
    ],
  ],
  [
    DatabaseCategory.OTHER,
    [
      'ChiTaRS',
      'EvolutionaryTrace',
      'GenomeRNAi',
      'GeneWiki',
      'NextBio',
      'PMAP-CutDB',
      'PRO',
    ],
  ],
]);

export default databaseCategoryToDatabases;
