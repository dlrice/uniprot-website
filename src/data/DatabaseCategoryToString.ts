import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

export default databaseCategoryToDatabases;
