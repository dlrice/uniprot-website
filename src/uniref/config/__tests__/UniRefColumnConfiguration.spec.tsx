import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import UniProtKBColumnConfiguration from '../UniRefColumnConfiguration';
import data from '../../__mocks__/UniRefResultsData.json';
import renderWithRedux from '../../../shared/__test-helpers__/RenderWithRedux';
jest.mock('../../../tools/utils/storage');

describe('UniRefColumnConfiguration component', () => {
  test('should render all columns', () => {
    UniProtKBColumnConfiguration.forEach((column) => {
      const { asFragment } = renderWithRedux(
        <MemoryRouter>{column.render(data)}</MemoryRouter>
      );
      expect(asFragment()).toMatchSnapshot(column.label);
    });
  });
});
