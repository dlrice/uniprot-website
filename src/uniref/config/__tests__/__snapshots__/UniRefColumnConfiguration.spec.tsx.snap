// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniRefColumnConfiguration component should render all columns: Cluster ID 1`] = `
<DocumentFragment>
  <a
    href="/uniref/undefined"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Cluster name 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render all columns: Common taxon 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render all columns: Common taxon ID 1`] = `
<DocumentFragment>
  <a
    href="/taxonomy/undefined"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Identity 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render all columns: Last updated 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render all columns: Length 1`] = `<DocumentFragment />`;

exports[`UniRefColumnConfiguration component should render all columns: Members 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Organism IDs 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Organisms 1`] = `
<DocumentFragment>
  <ul
    class="no-bullet"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Reference sequence 1`] = `
<DocumentFragment>
  <span
    class="break-anywhere"
  />
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Size 1`] = `
<DocumentFragment>
   member
</DocumentFragment>
`;

exports[`UniRefColumnConfiguration component should render all columns: Types 1`] = `<DocumentFragment />`;
